### PR TITLE
Add a way to restrict access to the search index

### PR DESCRIPTION
**Pull request by TPXP**
This enables developpers to restrict access to a search index by providing the `auth` parameter. This parameter is a function that takes as argument a userId (or null if the user is not connected), and returns `true` is the user is allowed to access the search index, or `false` if not allowed.

### CODE REVIEW

**Comment by TPXP** [comment]
I know I should add tests, but I don't know how to write them. If you insist on it, I can try, but I would prefer you to do it. :wink:

File: lib/easy-search-common.js [hunk 1]
```diff
@@ -63,9 +66,12 @@ EasySearch = (function () {
         publishScope = this,
         resultIds = [],
         publishHandle;
-
+
```
Comment (matteodem) [inline lib/easy-search-common.js]: please revert this line

**Comment by matteodem** [comment]
Awesome work! I'll write tests for in the next days :+1:

**Comment by matteodem** [comment]
I'll do some changes to your PR and then release 1.6.0

**Comment by matteodem** [comment]
I called it `permission` instead of auth (since you are not authenticating the user but rather restricting his access / permission to the index) and added some tests.

**Comment by btbjosh** [comment]
I'm trying to use permission with alanning:roles to do a `Roles.userIsInRole(Meteor.user(), 'admin')` in the permissions function but I don't know what argument to pass in for the user context. The first post in this thread says that the *permission* function accepts `userId` but I can't get that working. Any tips?

**Pull request merged**